hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -4279,6 +4307,50 @@ public Object load(Serializable id, Object optionalObject, LockOptions lockOptio
 		return loader.load( id, optionalObject, session, lockOptions );
 	}
 
+	@Override
+	public Object initializeEnhancedEntityUsedAsProxy(
+			Object entity,
+			String nameOfAttributeBeingAccessed,
+			SharedSessionContractImplementor session) {
+		final BytecodeEnhancementMetadata enhancementMetadata = getEntityMetamodel().getBytecodeEnhancementMetadata();
+		final BytecodeLazyAttributeInterceptor currentInterceptor = enhancementMetadata.extractLazyInterceptor( entity );
+		if ( currentInterceptor instanceof EnhancementAsProxyLazinessInterceptor ) {
+			final EnhancementAsProxyLazinessInterceptor proxyInterceptor = (EnhancementAsProxyLazinessInterceptor) currentInterceptor;
+
+			readLockLoader.load(
+					proxyInterceptor.getEntityKey().getIdentifier(),
+					entity,
+					session,
+					LockOptions.READ
+			);
+
+			final LazyAttributeLoadingInterceptor interceptor = enhancementMetadata.injectInterceptor(
+					entity,
+					proxyInterceptor.getEntityKey().getIdentifier(),
+					session
+			);
```

Comment:
IIUC ```PojoEntityTuplizer#afterInitialize``` has already injected the ```LazyAttributeLoadingInterceptor```. I don't think it hurts anything, but seems redundant.